### DISCUSSION TITLE
prevent extraction of images from pdf

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#3985] [Digital Image Processing]
Version: ImageMagick 6.9.10-71 Q16 x86_64 2021-07-06 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC OpenMP(3.1)
Delegates (built-in): bzlib fftw fontconfig freetype jng jp2 jpeg lzma openexr pangocairo png tiff x xml zlib

Centos 7.6

Is there a way to stop extracting of images from a pdf file. If any of the programs like pdfimages or pdftoppm are used, then each page should be extracted as an image but not images inside the page

Thanks

**Comment by snibgo** [comment]
Your question doesn't seem to be about ImageMagick.

ImageMagick will (via Ghostscript) rasterize each PDF page as a raster image. With policy.xml, you can prevent IM from reading PDF files.

**Comment by fmw42** [comment]
ImageMagick does not extract images from PDF files. It simply rasterizes everything on a page by page basis. That include all images on the page and all text on the page into one rasterized image containing everything that was on the page. Does that answer your question? If not, then please clarify further.

**Reply by srini8080** [reply]
I think it does. 

So once all images on the page and all text on the page into one rasterized image containing everything that was on the page, if someone tries to extract images from that page using some tools, they will be getting the entire page only as image and not sub-images right?

Which command should I use to rasterize the pdf page into an image and back to a pdf page?

Thanks

**Comment by snibgo** [comment]
The following rasterizes all the pages of a PDF into separate images. and writes them as pages to another PDF:
```
magick in.pdf out.pdf
```
To convert just the first page:
```
magick in.pdf[0] out.pdf
```

**Reply by srini8080** [reply]
Hmmm. That simple.

I guess sometimes thinking complex will just eat away time ;-)

Thanks much

**Reply by Jmuccigr** [reply]
What's the point of this? Someone has your PDF that has images in it, but you don't want them to have the images that they already have?

**Comment by srini8080** [comment]
;-)
